diff --git a/kernel/src/wasm/linker.rs b/kernel/src/wasm/linker.rs
index 1c94b7f..1a9e701 100644
--- a/kernel/src/wasm/linker.rs
+++ b/kernel/src/wasm/linker.rs
@@ -375,10 +375,10 @@ mod tests {
         let mut linker = WasmComponentLinker::new();

         // The `test_wasi.wasm` includes the cranelift integration
-        let wasm_bytes = include_bytes!("../../../test_wasi/target/wasm32-wasip1/debug/test_wasi.wasm");
+        let wasm_bytes = include_bytes!("../../../test_wasi/test_wasi.wasm");

-        let result = linker.link(wasm_bytes.to_vec());
+        let result = linker.add_module("test_wasi", wasm_bytes);
         // Depending on whether we want to run it, linking itself shouldn't fail due to imports
         assert!(result.is_ok());
     }
